### PR TITLE
Remove methods from /api/devfile that allow creating a workspace

### LINKED ISSUE #13867: Deprecate and remove /api/devfile method
### Is your task related to a problem? Please describe.
This is an outdated method to create a workspace. The correct one is ```/api/workspace/devfile```
And some users might be confused when using it.  Like this onehttps://github.com/eclipse/che/issues/13860

### Describe the solution you'd like

We should.
- Deprecate it with some good message first.
- Remove DevfileService completely

### Describe alternatives you've considered
none

### Additional context
none

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Remove methods from /api/devfile that allow creating a workspace

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/13867

#### Release Notes
n/a

#### Docs PR
n/a

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ci-test

**Comment by nickboldt** [comment]
Looks like this broke a lot of tests, which may need to be updated/removed too. :(

**Comment by skabashnyuk** [comment]
ci-test

**Review by mshaposhnik — APPROVED** [review]

**Comment by skabashnyuk** [comment]
ci-test

**Comment by skabashnyuk** [comment]
@eclipse/eclipse-che-qa can you comment on test results. Are they acceptable?

File: wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java [hunk 1]
```diff
@@ -13,59 +13,31 @@
...
 import org.eclipse.che.api.workspace.server.devfile.schema.DevfileSchemaProvider;
-import org.eclipse.che.api.workspace.shared.dto.WorkspaceDto;
-import org.eclipse.che.api.workspace.shared.dto.devfile.DevfileDto;
 
 @Api(value = "/devfile", description = "Devfile REST API")
 @Path("/devfile")
 public class DevfileService extends Service {
 
-  private WorkspaceLinksGenerator linksGenerator;
+  public static final String GET_SCHEMA_METHOD = "getSchema";
```
Comment (sleshchenko) [inline wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/devfile/DevfileService.java]: This is not used in this class.
Do you think it makes sense to add java docs and explain what is a value (like java method name) 
or move it where it's used, like DevfilePermissionsFilter. I prefer the second option.

**Review by sleshchenko — COMMENTED** [review]
Are you going to remove outdated and used `DevfileManager#exportWorkspace`, `DevfileManager#createWorkspace`, `DevfileManager#createWorkspaceConfig` methods? Or it's supposed in be done in scope of another issue?

**Comment by SkorikSergey** [comment]
> @eclipse/eclipse-che-qa can you comment on test results. Are they acceptable?

Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1951//Selenium_20tests_20report/) doesn't show any regression against this Pull Request.

**Comment by skabashnyuk** [comment]
ci-test

**Comment by skabashnyuk** [comment]
ci-test

**Comment by skabashnyuk** [comment]
>Are you going to remove outdated and used DevfileManager#exportWorkspace, DevfileManager#createWorkspace, DevfileManager#createWorkspaceConfig methods?

@sleshchenko these methods have been removed.

**Review by sleshchenko — APPROVED** [review]
LGTM

**Pull request merged**